File: Components/DevExpressVCL/Demos/ExpressEditors Library/CBuilder/EditorsStylesDemo/EditorsStylesDemoIssues.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "EditorsStylesDemoIssues.h"
#include "EditorsStylesDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxContainer"
#pragma link "cxControls"
#pragma link "cxEdit"
#pragma link "cxMemo"
#pragma link "cxPropertiesStore"
#pragma link "cxTextEdit"
#pragma link "EditorsStylesDemoBase"
#pragma link "cxCheckComboBox"
#pragma link "cxCheckListBox"
#pragma link "cxClasses"
#pragma link "cxDBCheckComboBox"
#pragma link "cxDBCheckListBox"
#pragma link "cxDBEdit"
#pragma link "cxDBLabel"
#pragma link "cxDBLookupComboBox"
#pragma link "cxDBLookupEdit"
#pragma link "cxDBProgressBar"
#pragma link "cxDBTrackBar"
#pragma link "cxDropDownEdit"
#pragma link "cxGroupBox"
#pragma link "cxLabel"
#pragma link "cxLookupEdit"
#pragma link "cxMaskEdit"
#pragma link "cxNavigator"
#pragma link "cxProgressBar"
#pragma link "cxSpinEdit"
#pragma link "cxStyles"
#pragma link "cxTrackBar"
#pragma link "cxDBNavigator"
#pragma resource "*.dfm"
TEditorsStylesDemoIssuesFrame *EditorsStylesDemoIssuesFrame;
//---------------------------------------------------------------------------
__fastcall TEditorsStylesDemoIssuesFrame::TEditorsStylesDemoIssuesFrame(TComponent* Owner)
  : TEditorsStylesDemoBaseFrame(Owner)
{
  HintStyle = hcstBlueSlideUp;
  FDisplayStyle = shtRainyDay;
  FTempDisplayStyle = shtRainyDay;
}
//---------------------------------------------------------------------------

void __fastcall TEditorsStylesDemoIssuesFrame::FillUsersCheckListBox()
{
  chlbUsers->Clear();
  EditorsStylesDemoDataDM->tblUsers->First();
  while (!EditorsStylesDemoDataDM->tblUsers->Eof) {
    chlbUsers->Items->Add()->Text =
        EditorsStylesDemoDataDM->tblUsersUserName->Value + " <" +
        EditorsStylesDemoDataDM->tblUsersEMAIL->Value + ">";
    EditorsStylesDemoDataDM->tblUsers->Next();
  }
  EditorsStylesDemoDataDM->tblItems->Refresh();
}
//---------------------------------------------------------------------------

void __fastcall TEditorsStylesDemoIssuesFrame::seFirstTargetPropertiesChange(
      TObject *Sender)
{
  pgbProgress->Properties->PeakValue = ((TcxDBSpinEdit*)Sender)->Value;
}
//---------------------------------------------------------------------------

String __fastcall __fastcall TEditorsStylesDemoIssuesFrame::Name()
{
  return "Issues database";
}
//---------------------------------------------------------------------------

String __fastcall __fastcall TEditorsStylesDemoIssuesFrame::BriefName()
{
  return "Issues";
}
//---------------------------------------------------------------------------

String TEditorsStylesDemoIssuesFrame::StylesIniPath()
{
  return "StylesFrmIssues\\";
}
//---------------------------------------------------------------------------

String TEditorsStylesDemoIssuesFrame::Description()
{
  return "Issues Database Notes";
}
//---------------------------------------------------------------------------

void __fastcall TEditorsStylesDemoIssuesFrame::seProgressPropertiesChange(
      TObject *Sender)
{
  seCheckProgress->Properties->MaxValue = seProgress->Value;
  if (seCheckProgress->Value > seProgress->Value)
    seCheckProgress->Value = seProgress->Value;
}
//---------------------------------------------------------------------------

void __fastcall TEditorsStylesDemoIssuesFrame::seCheckProgressPropertiesChange(
      TObject *Sender)
{
  pgbProgress->Properties->OverloadValue = seCheckProgress->Value;
}
//---------------------------------------------------------------------------


void __fastcall TEditorsStylesDemoIssuesFrame::FormShow(TObject *Sender)
{
  FillUsersCheckListBox();
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressEditors Library/Delphi/EditorsStylesDemo/EditorsStylesDemoIssues.pas
unit EditorsStylesDemoIssues;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, Menus, cxPropertiesStore, cxContainer, cxEdit, StdCtrls, ComCtrls,
  cxDropDownEdit, cxFontNameComboBox, cxDBFontNameComboBox, cxCheckComboBox,
  cxDBCheckComboBox, cxCheckListBox, cxDBCheckListBox, cxTrackBar,
  cxDBTrackBar, cxProgressBar, cxDBProgressBar, cxTextEdit, cxMaskEdit,
  cxColorComboBox, cxDBColorComboBox, cxControls, cxLabel, cxDBLabel,
  cxSplitter, cxNavigator, ExtCtrls, cxLookupEdit, cxDBLookupEdit,
  cxDBLookupComboBox, DB, cxSpinEdit, cxDBEdit, cxGroupBox,
  EditorsStylesDemoBase, cxFilterControl, cxSpinButton, cxMemo, cxStyles,
  cxCustomData, cxGraphics, cxFilter, cxData, cxDBData, cxClasses,
  cxDBNavigator;

type
  TEditorsStylesDemoIssuesFrame = class(TEditorsStylesDemoBaseFrame)
    Panel2: TPanel;
    lblDBIssueID: TcxDBLabel;
    chcbIDEs: TcxDBCheckComboBox;
    lbProject: TcxLabel;
    lblIssueID: TcxLabel;
    edProject: TcxDBLookupComboBox;
    lbCreator: TcxLabel;
    lbOwner: TcxLabel;
    edCreator: TcxDBLookupComboBox;
    edOwner: TcxDBLookupComboBox;
    gbStatus: TcxGroupBox;
    tbStatus: TcxDBTrackBar;
    lbNew: TcxLabel;
    lbPostponed: TcxLabel;
    lbFixed: TcxLabel;
    lbRejected: TcxLabel;
    gbProgress: TcxGroupBox;
    pgbProgress: TcxDBProgressBar;
    seProgress: TcxDBSpinEdit;
    lbIDEs: TcxLabel;
    Panel1: TPanel;
    lbIssue: TcxLabel;
    edIssue: TcxDBTextEdit;
    pnlNotification: TPanel;
    lbNotification: TcxLabel;
    chlbUsers: TcxDBCheckListBox;
    lbCheckProgress: TcxLabel;
    lbProgress: TcxLabel;
    seCheckProgress: TcxDBSpinEdit;
    cxLabel1: TcxLabel;
    seFirstTarget: TcxDBSpinEdit;
    cxDBNavigator1: TcxDBNavigator;
    procedure seCheckProgressPropertiesChange(Sender: TObject);
    procedure seProgressPropertiesChange(Sender: TObject);
    procedure seFirstTargetPropertiesChange(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    procedure FillUsersCheckListBox;
  public
    constructor Create(AOwner: TComponent); override;
    function Name: string; override;
    function BriefName: string; override;
    function StylesIniPath: string; override;
    function Description: String; override;
  end;

var
  EditorsStylesDemoIssuesFrame: TEditorsStylesDemoIssuesFrame;

implementation

uses EditorsStylesDemoData;

{$R *.dfm}

{ TEditorsStylesDemoIssuesFrame }

constructor TEditorsStylesDemoIssuesFrame.Create(AOwner: TComponent);
begin
  inherited Create(AOwner);
  HintStyle := hcstBlueSlideUp;
  FDisplayStyle := shtRainyDay;
  FTempDisplayStyle := shtRainyDay;
end;

procedure TEditorsStylesDemoIssuesFrame.FillUsersCheckListBox;
begin
  chlbUsers.Clear;
  with EditorsStylesDemoDataDM do
  begin
    tblUSERS.First;
    while not tblUSERS.Eof do
    begin
      chlbUsers.Items.Add.Text := string(tblUSERSUserName.Value + ' <' + tblUSERSEMAIL.Value + '>');
      tblUSERS.Next;
    end;
    tblITEMS.Refresh;
  end;
end;

function TEditorsStylesDemoIssuesFrame.Name: string;
begin
  Result := 'Issues database';
end;

function TEditorsStylesDemoIssuesFrame.BriefName: string;
begin
  Result := 'Issues';
end;

procedure TEditorsStylesDemoIssuesFrame.seCheckProgressPropertiesChange(
  Sender: TObject);
begin
  pgbProgress.Properties.OverloadValue := seCheckProgress.Value;
end;

procedure TEditorsStylesDemoIssuesFrame.seProgressPropertiesChange(
  Sender: TObject);
begin
  seCheckProgress.Properties.MaxValue := seProgress.Value;
  if seCheckProgress.Value > seProgress.Value then
    seCheckProgress.Value := seProgress.Value;
end;

function TEditorsStylesDemoIssuesFrame.StylesIniPath: string;
begin
  Result := 'StylesFrmIssues\';
end;

procedure TEditorsStylesDemoIssuesFrame.seFirstTargetPropertiesChange(
  Sender: TObject);
begin
  pgbProgress.Properties.PeakValue := TcxDBSpinEdit(Sender).Value;
end;

function TEditorsStylesDemoIssuesFrame.Description: String;
begin
  Result := 'Issues Database Notes';
end;

procedure TEditorsStylesDemoIssuesFrame.FormShow(Sender: TObject);
begin
  FillUsersCheckListBox;
end;

initialization
  EditorsStylesDemoFrameManager.RegisterFrameClass(TEditorsStylesDemoIssuesFrame);

end.
